presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrderCurveSearcher.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.zorder;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+/**
+ * This class contains functions to search for addresses within given value ranges.
+ */
+public class ZOrderCurveSearcher
+{
+    private final int[] dimensions;
+
+    private ZOrder zOrder;
+
+    private List<Integer> minimumValues;
+    private List<Integer> maximumValues;
+
+    private boolean returnLong;
+    private LinkedList<ZAddressRange> addressRanges;
+
+    /**
+     * Class constructor specifying the ZOrder class and initializing <code>dimensions</code>.
+     * <p/>
+     * The <code>dimensions</code> array stores how many values are stored at each level of the Z-curve.
+     */
+    public ZOrderCurveSearcher(ZOrder zOrder)
+    {
+        this.zOrder = zOrder;
+
+        // Create mutable version of encodingBits
```

Comment:
I think this comment is only for Line 48, right? But it could make the reader think the entire block is doing that. Please
1. Remove the comment.
2. Create function to initialize `dimension` member variable.